### ISSUE TITLE
Only homepage caches, and Let's Encrypt certification conflict with EE 4 RC1

### ISSUE THREAD

**Issue by Endda** [EasyEngine/easyengine#1272]
If you feel the issue is an EasyEngine core specific issue, please attach the output of the following commands.

System Information
- No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic
- EE 4.0.0-rc.1
- +-------------------+----------------------------------------------------------+
| OS                | Linux 4.15.0-23-generic #25-Ubuntu SMP Wed May 23 18:02: |
|                   | 16 UTC 2018 x86_64                                       |
| Shell             | /bin/bash                                                |
| PHP binary        | /usr/bin/php7.2                                          |
| PHP version       | 7.2.11-3+ubuntu18.04.1+deb.sury.org+1                    |
| php.ini used      | /etc/php/7.2/cli/php.ini                                 |
| EE root dir       | phar://ee.phar                                           |
| EE vendor dir     | phar://ee.phar/vendor                                    |
| EE phar path      | /root                                                    |
| EE packages dir   |                                                          |
| EE global config  | /opt/easyengine/config/config.yml                        |
| EE project config |                                                          |
| EE version        | 4.0.0-rc.1                                               |
+-------------------+----------------------------------------------------------+
- wp: command not found

_______________________________________________________________

So I wanted to test out RC1 last night and did so on a fresh install of Wordpress.

I used the command. . .

 - ee site create  shaylahleigh.com --type=wp --cache --admin-user=user--admin-pass=pass --ssl=le

to create the site and then just published some test posts. The NGINX Helper plugin was installed and activated, and the WP Redis plugin was installed and activated (which they should be, I presume).

However, only my homepage was being cached.

Opening up a Chrome incognito tab and going to the website ShaylahLeigh.com (was the URL I tested this install on), and then opened up the Chrome console before switching to the network tab.

I saw HIT/BYPASSon the home page (which indicates a cached page), but any other post that I went to (even when switching back and forth between the homepage and any of the four posts) showed BYPASS/BYPASS to indicate that the page wasn't being cached for some reason.

________________________________________________________________________

The second issue brought up in the title has to do with the Let's Encrypt certification. With EasyEngine 3, I could create my website using its compatible command listed above, and then go into Settings -> General and change the website URL from just blah.com to www.blah.com.

I have 7 or so websites up on EasyEngine 3 using this method and the Let's Encrypt certification works on the www. version just like it works on the non-www. version (Even though WordPress does a redirect to the www. version.

However, as you can see from going to ShaylayLeigh.com I am getting told the following error. . .

"This server could not prove that it is www.shaylahleigh.com; its security certificate is from shaylahleigh.com. This may be caused by a misconfiguration or an attacker intercepting your connection."

**Comment by caaol** [comment]
Have you ever tested admin-tools?  When I visited http://xx.com/ee-admin, I made an authentication, but I jumped to the front page.

**Comment by caaol** [comment]
And my other post is showed BYPASS/BYPASS

**Comment by caaol** [comment]
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial

ee cli info
+-------------------+-----------------------------------------------------------------------------------+
| OS                | Linux 4.11.8-041108-generic #201706290836 SMP Thu Jun 29 12:38:45 UTC 2017 x86_64 |
| Shell             | /bin/bash                                                                         |
| PHP binary        | /usr/bin/php7.2                                                                   |
| PHP version       | 7.2.11-3+ubuntu16.04.1+deb.sury.org+1                                             |
| php.ini used      | /etc/php/7.2/cli/php.ini                                                          |
| EE root dir       | phar://ee.phar                                                                    |
| EE vendor dir     | phar://ee.phar/vendor                                                             |
| EE phar path      | /root                                                                             |
| EE packages dir   |                                                                                   |
| EE global config  | /opt/easyengine/config/config.yml                                                 |
| EE project config |                                                                                   |
| EE version        | 4.0.0-rc.1                                                                        |
+-------------------+-----------------------------------------------------------------------------------+
wp --allow-root --info
-bash: wp: command not found

**Comment by caaol** [comment]
The first issue，When you set up permalink, the problem may be solved, and I have solved it.

**Comment by Endda** [comment]
> Have you ever tested admin-tools? When I visited http://xx.com/ee-admin, I made an authentication, but I jumped to the front page.

I'm not sure what the admin tools are.

Going to shaylahleigh.com/ee-admin and ignoring the SSL error, I get the following error. . .

"www.shaylahleigh.com redirected you too many times.
Try clearing your cookies.
ERR_TOO_MANY_REDIRECTS"

But I can overwrite that setting by making an edit to the wp-config.php file hardcoding the URL without the www part.

So after finding it (EE3 had sites at /var/www/ but now they are in /var/lib/docker/volumes/ (that took a while to find). So I put these lines in wp-config.php to bypass the www/SSL issue I was having

define('WP_HOME','http://shaylahleigh.com');
define('WP_SITEURL','http://shaylahleigh.com');

and now https://ShaylahLeigh.com is accessible. I can go and login like normal and remove the www. part from Settings -> General

**Comment by Endda** [comment]
> The first issue，When you set up permalink, the problem may be solved, and I have solved it.

I thought it may have been a permalink issue before, so last night I changed it /%postname%/ but I was still getting BYPASS/BYPASS on individual posts.

and I have no idea what ee-admin is. I get a 404 when trying to go to https://shaylahleigh.com/ee-admin

**Comment by Endda** [comment]
Should the --ssl=le not work on www prefixes like it did on EE3?

Do I need to do something like 

 - ee site create www.shaylahleigh.com --type=wp --cache --admin-user=user--admin-pass=pass --ssl=le

with EE4? because that didn't do anything with EE3 so that's why I didn't try it with EE4

**Comment by mrrobot47** [comment]
>and I have no idea what ee-admin is. I get a 404 when trying to go to https://shaylahleigh.com/ee-admin

@Endda till the documentation is fully released. Please check `ee help admin-tools`. You can set it up for a site using `ee admin-tools enable <site-name>`, then only will you get them on the url: http://example.com/ee-admin/

**Comment by Endda** [comment]
> > Do I need to do something like
> > ee site create [www.shaylahleigh.com](http://www.shaylahleigh.com) --type=wp --cache --admin-user=user--admin-pass=pass --ssl=le
> 
> @Endda Yes, that is correct. Please refer the following issue and the specified comment that was the final decision that has been implemented: [#1021 (comment)](https://github.com/EasyEngine/easyengine/issues/1021#issuecomment-393842656)
> 
> This will soon be released in the documentation.
> CC: @kirtangajjar

Oh okay. thank you for that information. I was always curious why it wasn't like that in EE3, but it worked by letting WordPress do the redirect, so that's cool.

**Comment by Endda** [comment]
> > and I have no idea what ee-admin is. I get a 404 when trying to go to https://shaylahleigh.com/ee-admin
> 
> @Endda till the documentation is fully released. Please check `ee help admin-tools`. You can set it up for a site using `ee admin-tools enable <site-name>`, then only will you get them on the url: http://example.com/ee-admin/

Ah, I will look into that either this weekend or after work on Monday. Thank you

**Comment by Endda** [comment]
So now that the website back up, is there any idea as to why I'm getting BYPASS/BYPASS instead of cache hits?

Do I need to install something other than NGINX Helper and WP Redis to get those HIT/BYPASS entries in Chrome's console?

Also, would I need to install another caching plugin to get a full suite of a cached website (browser cache, page cache, db cache, img/css/js/etc cache)? Or is EE4's cache option handling all of that for me?

If not would something like WP Rocket, W3 Total Cache or WP Super Cache be the go-to choice here? Again, if not, what would be recommended to compliment WP Redis with EE4 to cache as much as possible?

**Comment by caaol** [comment]
@mrrobot47 But  http://example.com/ee-admin/ can not visit.

**Comment by mrrobot47** [comment]
@Endda Can you share the url that you're hitting. You can obfuscate out the domain name. I would just like to know the entire url/query string with it if any.

>Do I need to install something
No. nginx-helper and WP Redis are already installed and configured for the site if `--cache` flag is passed.

**Comment by mrrobot47** [comment]
>@mrrobot47 But http://example.com/ee-admin/ can not visit.


@caaol can you be more specific. What is the issue that you're facing right now?

**Comment by Endda** [comment]
> @Endda Can you share the url that you're hitting. You can obfuscate out the domain name. I would just like to know the entire url/query string with it if any.
> 
> > Do I need to install something
> > No. nginx-helper and WP Redis are already installed and configured for the site if `--cache` flag is passed.

I have included it in this thread. These are the domains I am hitting in Chrome's incognito tab (which worked just fine to show me HIT/BYPASS messages with EE3

 - https://shaylahleigh.com/
 - https://shaylahleigh.com/testsfnadnfksdf/
 - https://shaylahleigh.com/testingss/
 - https://shaylahleigh.com/tests/
 - https://shaylahleigh.com/hello-world/

Last night I was able to get HIT/BYPASS on the homepage of just https://shaylahleigh.com/ but testing today showed that as BYPASS/BYPASS.

In any case, both last night and today I am getting BYPASS/BYPASS on all of those pages in Chrome's incognito tab using Chrome's network tab in its console

**Comment by Endda** [comment]
Trying to do some tests with Pingdom and GTMetrix and GTM is giving me the following error

 - An error occurred fetching the page: HTTPS error: hostname verification failed

While Google Pagespeed Insights says

 - Fetching the page exceeded the maximum of 10 redirects

So I may need to delete the test site and set it up again with www. and ssl=le for better testing

**Comment by mrrobot47** [comment]
>So I may need to delete the test site and set it up again with www. and ssl=le for better testing
Yes.

Also, don't trust browsers a lot.

```➜  ~ curl -XGET -IL https://shaylahleigh.com/tests/
HTTP/1.1 200 OK
Server: nginx
Date: Sat, 03 Nov 2018 16:53:18 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 56955
Connection: keep-alive
Vary: Accept-Encoding
X-Pingback: https://shaylahleigh.com/xmlrpc.php
Link: <https://shaylahleigh.com/wp-json/>; rel="https://api.w.org/"
Link: <https://shaylahleigh.com/?p=5>; rel=shortlink
X-SRCache-Fetch-Status: HIT
X-SRCache-Store-Status: BYPASS
X-Powered-By: EasyEngine v4.0.0-rc.1
```

**Comment by caaol** [comment]
@mrrobot47 I used  ee admin-tools enable wxmovie.club install admin tools, When I visited https://wxmovie.club/ee-admin/ , It still stays on my front page，and https://wxmovie.club/ee-admin/phpinfo.php ，File not found.

**Comment by Endda** [comment]
> > So I may need to delete the test site and set it up again with [www](http://www). and ssl=le for better testing
> > Yes.
> 
> Also, don't trust browsers a lot.
> 
> ```
> HTTP/1.1 200 OK
> Server: nginx
> Date: Sat, 03 Nov 2018 16:53:18 GMT
> Content-Type: text/html; charset=UTF-8
> Content-Length: 56955
> Connection: keep-alive
> Vary: Accept-Encoding
> X-Pingback: https://shaylahleigh.com/xmlrpc.php
> Link: <https://shaylahleigh.com/wp-json/>; rel="https://api.w.org/"
> Link: <https://shaylahleigh.com/?p=5>; rel=shortlink
> X-SRCache-Fetch-Status: HIT
> X-SRCache-Store-Status: BYPASS
> X-Powered-By: EasyEngine v4.0.0-rc.1
> ```

I mean, if that HIT/BYPASS is correct in the browser but the browser is just reporting it incorrectly. Then that's cool. But if that HIT/BYPASS indicator is only showing up in a terminal and the browser isn't caching anything, then that's an issue.

Getting HIT/BYPASS from a terminal doesn't help anything if the Browser is ignoring the cache (which could be why it's showing BYPASS/BYPASS). People don't browse websites with a terminal CURL command.

But I understand it's RC1 beta and all. I will remake the site with www. and let's encrypt later today. Thank you for your help.

**Comment by mrrobot47** [comment]
@caaol thanks for reposrting that the admin-tools url is not working properly and redirecting in wrong manner to home page. Can you please create an issue for the same here: https://github.com/EasyEngine/admin-tools-command
It will be looked into soon and resolved by `rc.2` or a newer nightly even sooner.

**Comment by mrrobot47** [comment]
@Endda here is the output from my chrome browser for your url.

![image](https://user-images.githubusercontent.com/25586785/47955112-02940900-dfb9-11e8-86d0-be0160df4d31.png)

It will work for sure in browser if it is working in terminal. It is just that it takes time and lots of opening and closing in case of browsers at times in testing. But stay assured the cache will work perfectly. Thanks a lot for testing. Keep on testing and reporting the issues you face :)

**Comment by Endda** [comment]
> It will be looked into soon and resolved by `rc.2` or a newer nightly even sooner.

Do you know if the "ee cli update" command will update our RC1 builds to the latest nightly, or is will that only work from RC1 to RC2 to RCX to Stable?

**Comment by mrrobot47** [comment]
Yes `ee cli update --nightly` will install the latest nightly.

```
ee cli update [--stable] [--nightly] [--yes]

  Default behavior is to check the releases API for the newest stable
  version, and prompt if one is available.

  Use `--stable` to install or reinstall the latest stable version.

  Use `--nightly` to install the latest built version of the develop branch.
  While not recommended for production, nightly contains the latest and
  greatest, and should be stable enough for development and staging
  environments.
```

**Comment by Endda** [comment]
> No. nginx-helper and WP Redis are already installed and configured for the site if `--cache` flag is passed.

But when I check the website in Google Pagespeed Insights. . .I am told that I need to "leverage browser cache" and lists files like. . .

 -     https://shaylahleigh.com/…ntyseventeen/assets/js/global.js?ver=1.0 (24 hours)
 -     https://shaylahleigh.com/…n/assets/js/jquery.scrollTo.js?ver=2.1.2 (24 hours)
 -     https://shaylahleigh.com/…assets/js/skip-link-focus-fix.js?ver=1.0 (24 hours)
 -     https://shaylahleigh.com/…emes/twentyseventeen/style.css?ver=4.9.8 (24 hours)
 -     https://shaylahleigh.com/…s/jquery/jquery-migrate.min.js?ver=1.4.1 (24 hours)
 -     https://shaylahleigh.com/…-includes/js/jquery/jquery.js?ver=1.12.4 (24 hours)
 -     https://shaylahleigh.com/…wp-includes/js/wp-embed.min.js?ver=4.9.8 (24 hours)
 -     https://shaylahleigh.com/…des/js/wp-emoji-release.min.js?ver=4.9.8 (24 hours)

So there are some things that aren't being cached in this current EE4 --cache method. Is there a preferred way of optimizing these?

**Comment by kirtangajjar** [comment]
> But when I check the website in Google Pagespeed Insights. I am told that I need to "leverage browser cache" and lists files like

Can you create another issue for this?

**Comment by kirtangajjar** [comment]
@Endda Closing this issue as it has been fixed in v4.0.
